XMonad/Hooks/EwmhDesktops.hs

```diff
@@ -171,10 +179,17 @@ ewmhDesktopsLogHookCustom t = withWindowSet $ \s -> do
         setNumberOfDesktops (length desktopNames)
         setDesktopNames desktopNames
 
-    -- Set client list; all windows, with focused windows last
-    let clientList = nub . concatMap (maybe [] (\(W.Stack x l r) -> reverse l ++ r ++ [x]) . W.stack) $ ws
+    -- Set client list which should be sorted by window age. We just
+    -- guess that StackSet contains windows list in this order which
+    -- isn't true but at least gives consistency with windows cycling
+    let clientList = nub . concatMap (reverse . W.integrate' . W.stack) $ ws
     whenChanged (ClientList clientList) $ setClientList clientList
 
+    -- Set stacking client list which should have bottom-to-top
+    -- stacking order, i.e. focused window should be last
+    let clientListStacking = nub . concatMap (maybe [] (\(W.Stack x l r) -> l ++ reverse r ++ [x]) . W.stack) $ ws
```

Comment:
What's the rationale for reversing the `r` here?

Reply:
Currently we have `reverse l ++ r ++ [x]` but it seemed to me incorrect. We should put there windows in ascending stacking and if we have for instance windows `1 2 [3] 4 5`, currently they will be reordered to `2 1 4 5 3` but it doesn't seem correct because 5 doesn't follow 3. After my fix they will be ordered as `1 2 5 4 3` - here after 3 we will have windows `4 5` which are next windows that we get with `focusDown`, and then windows `2 1` which we will get with `focusUp`.

Reply:
Hm, I think I misunderstood it a bit. StackSet stores `left` part in a reversed order so initially this reverse just restored windows ordering. So to follow my logic I should reverse left part as well. Or I can just leave it as it was before :)

Reply:
Changed it to what it was before - nobody complained about it and I don't have good explanation why it should be changed :)